hil/ext/switches/ovs.py

```diff
@@ -0,0 +1,244 @@
+        """
+        try:
+            for command in command_strings:
+                arg_list = shlex.split(command)
+                subprocess.check_call(arg_list)
+        except subprocess.CalledProcessError as e:
+            logger.error('%s', e)
+            raise SwitchError('ovs command failed: ')
+
+    def get_port_networks(self, ports):
+
+        response = {}
+        for port in ports:
+            trunk_id_list = self._interface_info(port.label)['trunks']
+            if trunk_id_list == []:
+                response[port] = []
+            else:
+                result = []
+                for trunk in trunk_id_list:
+                    name = "vlan/"+trunk
+                    result.append((name, trunk))
+                response[port] = result
+            native = self._interface_info(port.label)['tag']
+            if native != []:
+                response[port].append(("vlan/native", native))
+
+        return response
+
+    def revert_port(self, port):
+
+        shell_cmd_1 = 'sudo ovs-vsctl del-port {port}'.format(port=port)
+        shell_cmd_2 = 'sudo ovs-vsctl add-port {switch} {port}'.format(
+                    switch=self.hostname, port=port
+                    ) + ' vlan_mode=native-untagged'
+        self.ovs_connect(shell_cmd_1, shell_cmd_2)
+
+    def modify_port(self, port, channel, new_network):
+
+        (port,) = filter(lambda p: p.label == port, self.ports)
+        interface = port.label
+
+        if channel == 'vlan/native':
+            if new_network is None:
+                return self._remove_native_vlan(interface)
+            else:
+                return self._set_native_vlan(interface, new_network)
+        else:
+            match = re.match(re.compile(r'vlan/(\d+)'), channel)
+            assert match is not None, "HIL passed an invalid channel to the" \
+                " switch!"
+            vlan_id = match.groups()[0]
+
+            if new_network is None:
+                return self._remove_vlan_from_port(interface, vlan_id)
+            else:
+                assert new_network == vlan_id
+                return self._add_vlan_to_trunk(interface, vlan_id)
+
+# 2. Private Methods:
+
+    def _string_to_list(self, a_string):
+        """Converts a string representation of list to list.
+        Args:
+            a_string: list output recieved as string.
+                e.g. Strings starting with '[' and ending with ']'
+        Returns: object of list type.
+                 Empty list is put as None.
+        """
+        if a_string == '[]':
+            return ast.literal_eval(a_string)
+        else:
+            a_string = a_string.replace("[", "['").replace("]", "']")
+            a_string = a_string.replace(",", "','")
+            a_list = ast.literal_eval(a_string)
+            a_list = [ele.strip() for ele in a_list]
+            return a_list
+
+    def _string_to_dict(self, a_string):
+        """Converts a string representation of dictionary
```

Comment:
please put an example of how a_string looks like. I can't figure out what's happening here with all those replacements without having a reference example to work with. Again, from the description you have provided I should be able to just call literal_eval on a_string.
```
>>> sample = '{"key": "value"}'
>>> type(sample)
<type 'str'>
>>> ast.literal_eval(sample)
{'key': 'value'}
>>> type(ast.literal_eval(sample))
<type 'dict'>
```

Reply:
I have provided sample strings in both functions. Checkout the next commit.